server/src/test/java/io/crate/planner/optimizer/rule/EliminateCrossJoinTest.java

```diff
@@ -175,8 +175,14 @@ public void test_build_graph_from_a_double_nested_join_reorder_and_rebuild_to_lo
             "  └ Collect[doc.a | [x] | true]"
         );
 
-        var invalidOrder = EliminateCrossJoin.reorder(joinGraph, List.of(a, c, b));
-        assertThat(invalidOrder).isNull();
+        var orderWithCrossJoin = EliminateCrossJoin.reorder(joinGraph, List.of(a, c, b));
```

Comment:
here we are calling the rule again, but since the new flag is added, can this happen for a real query?

Reply:
yes, but the `reorder` is only called once in the `rule.apply()`, or not?

Reply:
Please look at the full test.

Reply:
Apologies, I was confused, and thought you're calling re-order on top of the result of the previous reorder. sorry.